hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -827,3 +833,49 @@ def draw_bytes(
     ) -> bytes:
         values = self._draw_collection(min_size, max_size, alphabet_size=2**8)
         return bytes(values)
+
+
+class URandom(Random):
+    # we reimplement a Random instance instead of using SystemRandom, because
+    # os.urandom is not guaranteed to read from /dev/urandom.
+
+    @staticmethod
+    def _urandom(size: int) -> bytes:
+        with open("/dev/urandom", "rb") as f:
+            return f.read(size)
+
+    def getrandbits(self, k: int) -> int:
+        assert k >= 0
+        size = bits_to_bytes(k)
+        n = int_from_bytes(self._urandom(size))
+        # trim excess bits
+        return n >> (size * 8 - k)
+
+    def random(self) -> float:
+        # adapted from random.SystemRandom.random
+        return (int_from_bytes(self._urandom(7)) >> 3) * (2**-53)
+
+
+class URandomProvider(HypothesisProvider):
+    # A provider which reads directly from /dev/urandom as its source of randomness.
+    # This provider exists to provide better Hypothesis integration with Antithesis
+    # (https://antithesis.com/), which interprets calls to /dev/urandom as the
+    # randomness to mutate. This effectively gives Antithesis control over
+    # the choices made by the URandomProvider.
+    #
+    # If you are not using Antithesis, you probably don't want to use this
+    # provider.
+
+    def __init__(self, conjecturedata: Optional["ConjectureData"], /):
+        super().__init__(conjecturedata)
+        if WINDOWS:  # pragma: no cover
```

Comment:
I was going to suggest a new [ignore pattern](https://github.com/HypothesisWorks/hypothesis/blob/c1a4d0361d79814c1d9b324ebd43a08969a7c053/hypothesis-python/.coveragerc#L21-L37) instead of this comment, but on reflection we can just wait for https://github.com/HypothesisWorks/hypothesis/issues/4261.

Reply:
ah, that would be nice...might add an ignore pattern anyway if I remember in the future.